rmf_site_editor/src/widgets/inspector/inspect_model_description/inspect_robot_properties.rs

```diff
@@ -15,27 +15,43 @@
  *
 */
 
-use super::{get_selected_description_entity, ModelDescriptionInspector};
+use super::{get_selected_description_entity, ModelDescriptionInspector, ModelPropertyQuery};
 use crate::{
     site::{
-        update_model_instances, Affiliation, ChangePlugin, Group, ModelMarker, ModelProperty, Robot,
+        update_model_instances, Change, ChangePlugin, Group, IssueKey, ModelMarker, ModelProperty,
+        NameInSite, Robot, SiteUpdateSet,
     },
     widgets::{prelude::*, Inspect},
-    ModelPropertyData,
+    AppState, Issue, ModelPropertyData, ValidateWorkspace,
+};
+use bevy::{
+    ecs::system::SystemParam,
+    prelude::{Component, *},
+    utils::Uuid,
 };
-use bevy::{ecs::system::SystemParam, prelude::*};
 use bevy_egui::egui::{ComboBox, Ui};
-use rmf_site_format::RobotProperty;
+use serde::{de::DeserializeOwned, Serialize};
+use serde_json::{Error, Map, Value};
 use smallvec::SmallVec;
 use std::collections::HashMap;
+use std::fmt::Debug;
+
+type InsertDefaultValueFn = fn() -> Result<Value, Error>;
 
-pub type ShowRobotPropertyWidgetFn = fn(&mut serde_json::Value, &mut Ui);
+pub struct RobotPropertyWidgetRegistration {
+    pub property_widget: Entity,
+    pub kinds: HashMap<String, RobotPropertyKindWidgetRegistration>,
+}
+
+pub struct RobotPropertyKindWidgetRegistration {
+    pub default: InsertDefaultValueFn,
+}
 
 /// This resource keeps track of all the properties that can be configured for a robot.
 #[derive(Resource)]
-pub struct RobotPropertyWidgets(pub HashMap<String, HashMap<String, ShowRobotPropertyWidgetFn>>);
+pub struct RobotPropertyWidgetRegistry(pub HashMap<String, RobotPropertyWidgetRegistration>);
```

Comment:
Small tip: When you have a [newtype](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) like this in bevy, if you add `#[derive(Deref, DerefMut)]` then you don't need to do `.0` to access the inner data. I.e. instead of

```rust
robot_property_widgets.0.get(property)
```

you can just do

```rust
robot_property_widgets.get(property)
```